pkg/auth/azure.go

```diff
@@ -152,7 +152,13 @@ func HttpAuthMiddleWare(resp http.ResponseWriter, req *http.Request, jwks *keyfu
 		}
 	}
 
-	err := ValidateToken(token, jwks, clientId, tenantId)
+	claims, err := ValidateToken(token, jwks, clientId, tenantId)
+
+	if _, ok := claims["tid"]; ok {
+		req.Header.Set("username", claims["name"].(string))
+		req.Header.Set("email", claims["email"].(string))
```

Comment:
email isn't a valid field. use preferred_username instead

Reply:
It is a valid field according to https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#payload-claims but it is true that preferred_username is a better option to use. Fixed!